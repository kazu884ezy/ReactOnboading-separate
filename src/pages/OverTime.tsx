import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { makeStyles } from "@material-ui/core/styles";
import {Grid} from "@material-ui/core";
import CurrencySelector from "../components/CurrencySelector";
import {
    setBaseCurrency,
    setTargetCurrency,
    getRatesOverTime,
    setOrder
} from "../actions/overtimeActions";
import RatesOverTimeTable from "../components/RatesOverTimeTable";
import Select from "@material-ui/core/Select";
import InputLabel from '@material-ui/core/InputLabel';
import FormControl from "@material-ui/core/FormControl";
import MenuItem from "@material-ui/core/MenuItem";

const useStyles = makeStyles({
    heading: {
        textAlign: 'center',
        fontSize:'24px',
    },
    main: {
        margin: '10px'
    },
    controls: {
        maxWidth: '1000px',
        margin: '10px auto'
    }
});

const OverTime = () => {
    const classes = useStyles();
    const dispatch = useDispatch();
    const ratesForPeriod = useSelector(state => state.overtime.ratesForPeriod);
    const baseCurrency = useSelector(state => state.overtime.baseCurrency);
    const targetCurrency = useSelector(state => state.overtime.targetCurrency);
    const order = useSelector(state => state.overtime.order);

    useEffect(() => {
        dispatch(getRatesOverTime(baseCurrency, targetCurrency, order === 'desc'));
    }, [dispatch, baseCurrency, targetCurrency, order])

    const handleCurrencyChange = currency => {
        dispatch(setBaseCurrency(currency));
    }

    const handleTargetCurrencyChange = currency => {
        dispatch(setTargetCurrency(currency))
    }

    const handleOrderChange = (event : React.ChangeEvent<{ value: unknown }>) => {
        dispatch(setOrder(event.target.value as string));
    }

    return (
        <div>
            <h1 className={classes.heading}>Exchange Rates Over Time</h1>
            <Grid container spacing={3} justify="center">
                <Grid item>
                    <CurrencySelector onChange={handleCurrencyChange} initialValue={baseCurrency} />
                </Grid>
                <Grid item>
                    <CurrencySelector onChange={handleTargetCurrencyChange} initialValue={targetCurrency} />
                </Grid>
            </Grid>
            <div className={classes.main}>
                <div className={classes.controls}>
                    <FormControl>
                        <InputLabel>Order Date:</InputLabel>
                        <Select
                            value={order}
                            onChange={handleOrderChange}
                        >
                            <MenuItem value='desc'>descending</MenuItem>
                            <MenuItem value='asc'>ascending</MenuItem>
                        </Select>
                    </FormControl>
                </div>
                <RatesOverTimeTable overtimeRates={ratesForPeriod} />
            </div>
        </div>
    );
};

export default OverTime;