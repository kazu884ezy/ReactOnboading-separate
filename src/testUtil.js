import store from './store';

export const makeTestStore = () => {
    const origDispatch = store.dispatch;
    store.dispatch = jest.fn(origDispatch);
    return store;
};