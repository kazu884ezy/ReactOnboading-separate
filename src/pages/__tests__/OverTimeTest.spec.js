import React from 'react';
import {Provider} from 'react-redux'
import { render} from '@testing-library/react';
import OverTime from "../OverTime";
import {makeTestStore} from "../../testUtil";

describe('OverTime page test', () => {
    test('render correctly', () => {
        const store = makeTestStore();

        const { container } = render(<Provider store={store}><OverTime /></Provider>);
        expect(container.firstChild).toMatchSnapshot();
        // screen.debug();
        // screen.queryByTestId('test1');
        // screen.getByRole('');
       // expect(screen.getByText('Page Not Found')).toBeInTheDocument();
    })

    // test('handleCurrencyChange', () => {
    //     const spy = jest.spyOn(OverTime, 'handleCurrencyChange');
    //     expect(spy.handleCurrencyChange('test')).toBe('test');
    // })

    // test('onChange', () => {
    //     const onChange = jest.fn();
    //     const store = makeTestStore();
    //
    //     render(<Provider store={store}><OverTime /></Provider>);
    //
    //     fireEvent.change(screen.getBy {
    //         target: {value: 'AUD'},
    //     })
    //
    //     expect(onChange).toHaveBeenCalledTimes(1);
    // })
})