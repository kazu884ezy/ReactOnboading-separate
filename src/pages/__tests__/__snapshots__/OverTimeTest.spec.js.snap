// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverTime page test render correctly 1`] = `
<div>
  <h1
    class="makeStyles-heading-1"
  >
    Exchange Rates Over Time
  </h1>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiFormControl-root"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="mui-component-select-base_currency"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
            id="mui-component-select-base_currency"
            role="button"
            tabindex="0"
          >
            NZD
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            name="base_currency"
            tabindex="-1"
            value="NZD"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiFormControl-root"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="mui-component-select-target_currency"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
            id="mui-component-select-target_currency"
            role="button"
            tabindex="0"
          >
            AUD
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            name="target_currency"
            tabindex="-1"
            value="AUD"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    class="makeStyles-main-2"
  >
    <div
      class="makeStyles-controls-3"
    >
      <div
        class="MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
        >
          Order Date:
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <div
            aria-haspopup="listbox"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
            role="button"
            tabindex="0"
          >
            descending
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            tabindex="-1"
            value="desc"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <table
      class="MuiTable-root makeStyles-table-4"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Date
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Value
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      />
    </table>
  </div>
</div>
`;
